src/client/app/actions/barReadings.js

```diff
@@ -5,47 +5,65 @@
 
 export const REQUEST_BAR_READINGS = 'REQUEST_BAR_READINGS';
 export const RECEIVE_BAR_READINGS = 'RECEIVE_BAR_READINGS';
 
 /**
  * @param {State} state
- * @param {number} meterID
+ * @param {number} dsID
  * @param {TimeInterval} timeInterval
  * @param {Moment.Duration} barDuration
+ * @param {String} dstype either DATA_TYPE_METER, DATA_TYPE_GROUP
  */
-function shouldFetchBarReadings(state, meterID, timeInterval, barDuration) {
-	const readingsForMeterID = state.readings.bar.byMeterID[meterID];
-	if (readingsForMeterID === undefined) {
+function shouldFetchBarReadings(state, dsID, timeInterval, barDuration, dstype) {
+	let readingsForID;
+	if (dstype === DATA_TYPE_METER) {
+		readingsForID = state.readings.bar.byMeterID[dsID];
+	} else if (dstype === DATA_TYPE_GROUP) {
+		readingsForID = state.readings.bar.byGroupID[dsID];
+	}
+	if (readingsForID === undefined) {
 		return true;
 	}
-	if (readingsForMeterID[timeInterval] === undefined) {
+	if (readingsForID[timeInterval] === undefined) {
 		return true;
 	}
-	if (readingsForMeterID[timeInterval][barDuration] === undefined) {
+	if (readingsForID[timeInterval][barDuration] === undefined) {
 		return true;
 	}
-	const readingsForTimeIntervalAndDuration = state.readings.bar.byMeterID[meterID][timeInterval][barDuration];
+	const readingsForTimeIntervalAndDuration = readingsForID[timeInterval][barDuration];
 	return readingsForTimeIntervalAndDuration === undefined && !readingsForTimeIntervalAndDuration.isFetching;
 }
 
-function requestBarReadings(meterIDs, timeInterval, barDuration) {
-	return { type: REQUEST_BAR_READINGS, meterIDs, timeInterval, barDuration };
+function requestBarReadings(dsIDs, timeInterval, barDuration, dstype) {
+	return { type: REQUEST_BAR_READINGS, dsIDs, timeInterval, barDuration, dstype };
 }
 
-function receiveBarReadings(meterIDs, timeInterval, barDuration, readings) {
-	return { type: RECEIVE_BAR_READINGS, meterIDs, timeInterval, barDuration, readings };
+function receiveBarReadings(dsIDs, timeInterval, barDuration, readings, dstype) {
+	return { type: RECEIVE_BAR_READINGS, dsIDs, timeInterval, barDuration, readings, dstype };
 }
 
-function fetchBarReadings(meterIDs, timeInterval) {
+function fetchBarReadings(dsIDs, timeInterval, dstype) {
 	return (dispatch, getState) => {
 		const barDuration = getState().graph.barDuration;
-		dispatch(requestBarReadings(meterIDs, timeInterval, barDuration));
-		const stringifiedMeterIDs = meterIDs.join(',');
-		return axios.get(`/api/readings/bar/${stringifiedMeterIDs}`, {
+		dispatch(requestBarReadings(dsIDs, timeInterval, barDuration, dstype));
+		const stringifiedIDs = dsIDs.join(',');
+
+		let endpoint;
+		if (dstype === DATA_TYPE_METER) {
+			endpoint = '/api/readings/bar/meters';
+		} else if (dstype === DATA_TYPE_GROUP) {
+			endpoint = '/api/readings/bar/groups';
+		} else {
+			console.error('Unknown datatype requested in fetchBarReadings: ', dstype);
+			endpoint = '/api/nonexistant';
```

Comment:
Perhaps just return Promise.resolve() here instead of sending an unnecessary request to a non-existant endpoint.